drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -336,6 +336,50 @@ def data_collection_features
     end.compact
   end
 
+  def occurrence_point_forms
+    # Get definitions for Occurrence Point forms, including inactive/retired (but excluding drafts)
+    definitions = Hmis::Form::Definition.with_role(:OCCURRENCE_POINT).published_or_retired.latest_versions
+
+    definitions.map do |definition|
+      # Choose the most specific instance for this enrollment
+      best_instance = definition.instances.active.detect_best_instance_for_enrollment(enrollment: self)
+
+      # If we didn't find a best_instance, check for legacy data
+      has_any_data = false
+      unless best_instance
+        definition.walk_definition_nodes(as_open_struct: true) do |item|
+          next unless item.mapping.present?
+
+          record_type = item.mapping&.record_type
+          field_name = item.mapping&.field_name&.underscore
+          custom_field_key = item.mapping&.custom_field_key
+
+          next unless record_type == 'ENROLLMENT' || custom_field_key
+
+          if record_type && field_name
+            has_any_data = respond_to?(field_name) && send(field_name).present?
+          elsif custom_field_key
+            # For simplicity, for now, just look for CDEDs where the owner is an Enrollment
+            owner_class_name = self.class.sti_name
+            cded = Hmis::Hud::CustomDataElementDefinition.for_type(owner_class_name).find_by(key: custom_field_key)
+            has_any_data = cded && Hmis::Hud::CustomDataElement.where(data_element_definition: cded, owner: self).any?
```

Comment:
I think we can pull these lookups out so we're not hitting the db so much. For example:
```
# enrollment cdeds that this enrollment has CDE record(s) for:
cdeds_this_enrollment_has = custom_data_element_definitions.pluck(:key).to_set

...

# in the definition/item loops, no need to hit the db:
has_any_data = cdeds_this_enrollment_has.include?(custom_field_key)

```

The `custom_data_element_definitions` relation comes from this concern:
https://github.com/greenriver/hmis-warehouse/blob/e566d8a7f0bf834f548b2dc5086b002210c5af98/drivers/hmis/app/models/hmis/hud/concerns/form_submittable.rb#L16-L17